diff --git a/drivers/md/dm-android-verity.c b/drivers/md/dm-android-verity.c
index 2b0cc969b..f93bfa5c5 100644
--- a/drivers/md/dm-android-verity.c
+++ b/drivers/md/dm-android-verity.c
@@ -118,9 +118,7 @@ static inline bool is_userdebug(void)
 
 static inline bool is_unlocked(void)
 {
-	static const char unlocked[] = "orange";
-
-	return !strncmp(verifiedbootstate, unlocked, sizeof(unlocked));
+	return true;
 }
 
 static int read_block_dev(struct bio_read *payload, struct block_device *bdev,
@@ -713,6 +711,11 @@ static int android_verity_ctr(struct dm_target *ti, unsigned argc, char **argv)
 	if (is_eng())
 		return create_linear_device(ti, dev, target_device);
 
+	if (is_unlocked()) {
+		DMINFO("Device unlocked, bypassing verity - using linear target");
+		return create_linear_device(ti, dev, target_device);
+	}
+
 	strreplace(key_id, '#', ' ');
 
 	DMINFO("key:%s dev:%s", key_id, target_device);
@@ -726,11 +729,6 @@ static int android_verity_ctr(struct dm_target *ti, unsigned argc, char **argv)
 	err = extract_metadata(dev, &fec, &metadata, &verity_enabled);
 
 	if (err) {
-		/* Allow invalid metadata when the device is unlocked */
-		if (is_unlocked()) {
-			DMWARN("Allow invalid metadata when unlocked");
-			return create_linear_device(ti, dev, target_device);
-		}
 		DMERR("Error while extracting metadata");
 		handle_error();
 		return err;
